consumer/logs.go

```diff
@@ -35,8 +35,17 @@ func NewLogs(consume ConsumeLogsFunc, options ...Option) (Logs, error) {
 	if consume == nil {
 		return nil, errNilFunc
 	}
+
+	baseImpl := newBaseImpl(options...)
+	fn := func(ctx context.Context, ld plog.Logs) error {
+		baseImpl.obsreport.StartTracesOp(ctx)
+		err := consume(ctx, ld)
+		baseImpl.obsreport.EndTracesOp(ctx, ld.LogRecordCount(), err)
```

Comment:
Worth looking at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29274.

It's arguably a problem with the API here but at least we should capture the count prior to calling ConsumeLogs.